### PR TITLE
ansible: added etcd cert auth support

### PR DESCRIPTION
**Pull request by kayrus**
subj

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1636)

### CODE REVIEW

File: ansible/roles/etcd/defaults/main.yaml [hunk 1]
```diff
@@ -9,8 +9,13 @@ etcd_peer_port: 2380
 etcd_client_legacy_port: 4001
 etcd_peer_legacy_port: 7001
 etcd_peers_group: etcd
+
 etcd_url_scheme: http
 etcd_peer_url_scheme: http
+# Set correspondingly to true when use https values above to enable cert auth
+etcd_client_cert_auth: true
+etcd_peer_client_cert_auth: true
```
Comment (ingvagabund) [inline ansible/roles/etcd/defaults/main.yaml]: Both ``ETCD_CLIENT_CERT_AUTH`` and ``ETCD_PEER_CLIENT_CERT_AUTH`` are false by default.

Setting both to ``true`` changes the default behaviour of deployed etcd. I would rather set it to false and leave the final decision to operator. What do you think?

[1] https://github.com/coreos/etcd/blob/master/Documentation/op-guide/configuration.md#--client-cert-auth
[2] https://github.com/coreos/etcd/blob/master/Documentation/op-guide/configuration.md#--peer-client-cert-auth

Comment (kayrus) [inline ansible/roles/etcd/defaults/main.yaml]: May be. But there should be a note somewhere that TLS cert doesn't imply the auth by cert.

Comment (ingvagabund) [inline ansible/roles/etcd/defaults/main.yaml]: Can you write that down into README.md?

Comment (kayrus) [inline ansible/roles/etcd/defaults/main.yaml]: Ok, little bit later.

**Comment by kayrus** [comment]
@ingvagabund Updated README.md

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Pull request merged**